cmd/controller-manager/app/options/debugging.go

```diff
@@ -27,6 +28,16 @@ type DebuggingOptions struct {
 	*componentbaseconfig.DebuggingConfiguration
 }
 
+// RecommendedDebuggingOptions returns the currently recommended debugging options.  These are subject to change
```

Comment:
to clarify: there is a commandline args to set this option for controllers, any particular reason this logic is added here to control it ?

Reply:
This is refer to community PR: https://github.com/kubernetes/kubernetes/pull/88663